Mathlib/Topology/Algebra/InfiniteSum/UniformOn.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2025 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck, David Loeffler
+-/
+import Mathlib.Topology.Algebra.InfiniteSum.Defs
+import Mathlib.Topology.Algebra.UniformConvergence
+
+/-!
+# Infinite sum and products that converge uniformly on a set
+
+This file defines the notion of uniform convergence of infinite sums and products of functions,
+on a given family of subsets of their domain.
+
+It also defines the notion of local uniform convergence of infinite sums and products of functions
+on a set.
+-/
+
+noncomputable section
+
+open Filter Function
+
+open scoped Topology
+
+variable {α β ι : Type*} [CommMonoid α]  {f : ι → β → α} {g : β → α} {𝔖 : Set (Set β)}
+  {x : β} {s : Set β} {I : Finset ι} [UniformSpace α]
+
+/-!
+## Uniform convergence of sums and products
+-/
+
+section UniformlyOn
+
+variable (f g 𝔖)
+
+/-- `HasProdUniformlyOn f g 𝔖` means that the (potentially infinite) product `∏' i, f i b`
+for `b : β` converges uniformly on each `s` in a family of sets `𝔖` to `g`. -/
+@[to_additive "`HasSumUniformlyOn f g 𝔖` means that the (potentially infinite) sum `∑' i, f i b`
+for `b : β` converges uniformly on `s ∈ 𝔖` to `g`."]
+def HasProdUniformlyOn : Prop :=
+  HasProd (fun i ↦ UniformOnFun.ofFun 𝔖 (f i)) (UniformOnFun.ofFun 𝔖 g)
+
+/-- `MultipliableUniformlyOn f 𝔖` means that `f` converges uniformly on `s` to some infinite
+product. -/
+@[to_additive "`SummableUniformlyOn f s` means that `f` converges uniformly on `s` to some
+infinite sum."]
```

Comment:
I tried to rephrase this in a way to ensure that its clear that the function to which the series converges does not depend on `s ∈ 𝔖`.

There should also be some comment (analogous to the one in the docstring of `Multipliable`) about using `tsum` to get the sum. Something like: "Use `fun x ↦ ∏' i, f i x` to get the product function."
```suggestion
/-- `MultipliableUniformlyOn f 𝔖` means that there is some infinite product to which
`f` converges uniformly on every `s ∈ 𝔖`. -/
@[to_additive "`SummableUniformlyOn f s` means that there is some infinite sum to 
which `f` converges uniformly on every `s ∈ 𝔖`."]
```